compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -603,7 +603,7 @@ trait Applications extends Compatibility { self: Typer with Dynamic =>
    * argument trees.
    */
   class ApplicableToTreesDirectly(methRef: TermRef, targs: List[Type], args: List[Tree], resultType: Type)(implicit ctx: Context) extends ApplicableToTrees(methRef, targs, args, resultType)(ctx) {
-    override def argOK(arg: TypedArg, formal: Type): Boolean = argType(arg, formal) <:< formal
+    override def argOK(arg: TypedArg, formal: Type): Boolean = argType(arg, formal) <:< formal.widenExpr
```

Comment:
Do we have some guarantee that the result of `argType` cannot be an ExprType here ? Even if we do, it seems more regular to widen both sides.

Reply:
The case we dropped in TypeComparer is where the RHS is an ExprType. `=> T` was never a subtype of `T`.

Reply:
Right, but if both `argType` and `formal` are `=> T`, then `argOK` used to return true but will return false now that we widen `formal`.

Reply:
I see. I believe argTypes are never ExprTypes. They might be TermRefs with an underlying ExprType. But we skip the ExprType when we do a widen on these. `lub` also does `widen`, not `widenExpr`. So I think we are good.